packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -117,14 +117,22 @@ export class MediaPlaceholder extends Component {
 			multiple,
 			onError,
 			onSelect,
-			value = [],
 		} = this.props;
 		let setMedia;
 		if ( multiple ) {
 			if ( addToGallery ) {
-				const currentValue = value;
+				let lastMediaPassed = [];
 				setMedia = ( newMedia ) => {
-					onSelect( currentValue.concat( newMedia ) );
+					const currentMedia = this.props.value || [];
+					// Remove lastMediaPassed from currentMedia.
+					const mediaToReturn = currentMedia.filter( ( item ) => {
+						return ! lastMediaPassed.some( ( temporaryMediaURL ) => temporaryMediaURL === item.url );
+					} );
+					// Reset lastMediaPassed and set it with urls from newMedia.
+					lastMediaPassed = [];
```

Comment:
Could we just do this if `lastMediaPassed` is just being set to all the urls in newMedia? might be a bit clearer

```suggestion
					lastMediaPassed = newMedia.map( ( media ) => media.url );
```

Reply:
Definitely!  Updated to do this =)